pkg/runctl/secret_manager_test.go

```diff
@@ -0,0 +1,186 @@
+package runctl
+
+import (
+	"fmt"
+	"testing"
+
+	stewardv1alpha1 "github.com/SAP/stewardci-core/pkg/apis/steward/v1alpha1"
+	mocks "github.com/SAP/stewardci-core/pkg/k8s/mocks"
+	secretMocks "github.com/SAP/stewardci-core/pkg/k8s/secrets/mocks"
+	gomock "github.com/golang/mock/gomock"
+	"gotest.tools/assert"
+)
+
+// TODO: write better matcher to check correct filters and transformers
+var (
+	pipelineSecretTransormerMatcher  = gomock.Len(2)
+	imagePullSecretFilterMatcher     = gomock.Any()
+	imagePullSecretTransormerMatcher = gomock.Len(4)
+	cloneSecretTransormerMatcher     = gomock.Len(4)
+
+	spec = &stewardv1alpha1.PipelineSpec{
+		JenkinsFile: stewardv1alpha1.JenkinsFile{
+			RepoAuthSecret: "scm_secret1",
+		},
+		Secrets: []string{
+			"secret1",
+			"secret2",
+		},
+		ImagePullSecrets: []string{
+			"imagePullSecret1",
+			"imagePullSecret2",
+		},
+	}
+)
```

Comment:
I don't like using global variables for test helpers. I would put them into a struct, and use a constructor function for initialization.

Reply:
See https://github.com/SAP/stewardci-core/pull/165/commits/4bda8a075cf98dcd166781f5b92c0fcb7e186d55